blocks/product-cards/product-cards.js

```diff
@@ -1,35 +1,35 @@
-import { getProducts } from "../../scripts/product.js";
-import createTag from "../../utils/tag.js";
+import { getProducts } from '../../scripts/product.js';
+import createTag from '../../utils/tag.js';
 
 export default async function decorate(block) {
-  block.replaceWith(createProductCards(await getProducts()));
+  const { searchParams } = new URL(window.location.href);
+  const categoryId = searchParams.get('category_id');
+  block.replaceWith(createProductCards(await getProducts(categoryId)));
```

Comment:
clicking on start shopping button will also bring us to products page and it wont give you category id. We should handle the case of no category id found here.